### ISSUE TITLE
expectEvent crashes when decoding conflicting events (identical signatures, differently indexed args)

### ISSUE THREAD

**Issue by 3esmit** [OpenZeppelin/openzeppelin-test-helpers#132]
Currently I run in this bug while trying to use expectEvent:

```
(node:26057) UnhandledPromiseRejectionWarning: Error: Returned values aren't valid, did it run Out of Gas? You might also see this error if you are not using the correct ABI for the contract you are retrieving data from, requesting data from a block number that does not exist, or querying a node which is not fully synced.
    at ABICoder.decodeParameters (/home/ricardo/Documents/GitHub/status-im/ens-usernames/node_modules/web3-eth-abi/src/index.js:239:15)
    at ABICoder.decodeLog (/home/ricardo/Documents/GitHub/status-im/ens-usernames/node_modules/web3-eth-abi/src/index.js:302:52)
    at logs.filter.map.log (/home/ricardo/Documents/GitHub/status-im/ens-usernames/node_modules/@openzeppelin/test-helpers/src/expectEvent.js:135:30)
    at Array.map (<anonymous>)
    at decodeLogs (/home/ricardo/Documents/GitHub/status-im/ens-usernames/node_modules/@openzeppelin/test-helpers/src/expectEvent.js:135:6)
    at Function.inTransaction (/home/ricardo/Documents/GitHub/status-im/ens-usernames/node_modules/@openzeppelin/test-helpers/src/expectEvent.js:96:16)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:26057) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 98)
(node:26057) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This is caused when the decoder tries to decode events of a transaction which have identical signature but different indexed parameters.

Consider the following use-case:

Contract that accept ERC20 tokens as payment and mints ERC721 tokens. 
When the ERC20 token is transfered to the contract, a event is emmited by the ERC20 token: `event Transfer(address indexed from, address indexed to, uint256 value);`
Then the contract mints the ERC721 token: `event Transfer(address indexed from, address indexed to, uint256 indexed tokenId);`

```solidity

// SPDX-License-Identifier: CC0-1.0

pragma solidity 0.6.2;

import "../openzeppelin-solidity/token/ERC721/ERC721.sol";
import "../openzeppelin-solidity/token/ERC20/ERC20.sol";


contract ERC20BuyERC721Mint is ERC721 {
    ERC20 public token;

    constructor(ERC20 _token) ERC721("TEST", "Test"){
        token = _token;
    }

    function buy(){
        token.transferFrom(msg.sender,address(this), 1);
        _mint(msg.sender, uint256(msg.sender));
    }
}
```

```js
     const buyResult = await ERC20BuyERC721Mint.methods.buy().send();
     expectEvent.inTransaction(buyResult.transactionHash, ERC20, 'Transfer') //errors here
```

This causes a bug in the expectEvent decoder.

**Comment by nventuro** [comment]
We're filtering events by topic, but it seems [the `indexed` attribute is *not* part of the event **signature](https://solidity.readthedocs.io/en/develop/abi-spec.html?highlight=canonical#events). How are tools supposed to distinguish these two cases?

**Comment by 3esmit** [comment]
By the emitter. The same emitter could in theory emit both, but Solidity won't compile if two events have the same signature.

**Comment by 3esmit** [comment]
Also, it could try catch both.

**Comment by 3esmit** [comment]
I submitted a PR to fix this issue on #133, using the "emitter" address check. To capture a conflicting event from other contract, inTransaction must be used and the contract passed must have an address. 

To capture a conflicting event in the called contract, nothing really changes.
